components/style/values/computed/border.rs

```diff
@@ -81,3 +84,56 @@ impl ToAnimatedZero for BorderCornerRadius {
         Err(())
     }
 }
+
+/// The computed value of the `border-image-repeat` property:
+///
+/// https://drafts.csswg.org/css-backgrounds/#the-border-image-repeat
+#[derive(Clone, Debug, MallocSizeOf, PartialEq)]
+pub struct BorderImageRepeat(pub RepeatKeyword, pub RepeatKeyword);
+
+impl BorderImageRepeat {
+    /// Returns the `repeat repeat` value.
+    pub fn repeat() -> Self {
+        BorderImageRepeat(RepeatKeyword::Repeat, RepeatKeyword::Repeat)
+    }
+}
+
+impl ToCss for BorderImageRepeat {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result
+    where
+        W: fmt::Write,
+    {
+        match (self.0, self.1) {
+            (horizontal, vertical) => {
+                horizontal.to_css(dest)?;
+                if horizontal != vertical {
+                    dest.write_str(" ")?;
+                    vertical.to_css(dest)?;
+                }
+                Ok(())
+            },
+        }
+    }
+}
+
+impl ToComputedValue for SpecifiedBorderImageRepeat {
+    type ComputedValue = BorderImageRepeat;
+
+    #[inline]
+    fn to_computed_value(&self, _: &Context) -> Self::ComputedValue {
+        match *self {
+            SpecifiedBorderImageRepeat::Keywords(horizontal, vertical) => {
+                BorderImageRepeat(horizontal, vertical.unwrap_or(horizontal))
+            }
+        }
```

Comment:
I'm wondering, when we only have one pattern, will it be better to use `if let` here?